### PR TITLE
Use stylesheet.css as minimal extra stylesheet

### PR DESCRIPTION
**Pull request by albert-github**
When there are only few changes in the stylesheet compared to the original stylesheet (in this case the 1.8.6 version) it is better to just place these changes in an extra stylesheet so that updates in the official doxygen stylesheet are automatically taken into account.
- adjusted the Doxyfile.in to reflect this
- changed the stylesheet.css so it only contains the real changes compared to the default style sheet.

### CODE REVIEW

**Comment by albert-github** [comment]
When the proposed pull request #637 is merged this proposed pull request is obsolete.

File: src/Doxyfile.in [hunk 1]
```diff
@@ -1130,7 +1130,7 @@ HTML_STYLESHEET        = @GUDHI_DOXYGEN_COMMON_DOC_PATH@/stylesheet.css
 # list). For an example see the documentation.
 # This tag requires that the tag GENERATE_HTML is set to YES.
 
-HTML_EXTRA_STYLESHEET  =
+HTML_EXTRA_STYLESHEET  = @GUDHI_DOXYGEN_COMMON_DOC_PATH@/stylesheet.css
```
Comment (VincentRouvreau) [inline src/Doxyfile.in:1133]: I don't know why, but this change makes the header drop down menus no more work:
[before](https://gudhi.inria.fr/doc/latest/)

Screenshot from 2022-06-08 09-52-25

[after](https://output.circle-artifacts.com/output/job/2af5cf4a-b80e-407a-accd-6429394e73d1/artifacts/0/doxygen/index.html) [may require a circleci acount]

Screenshot from 2022-06-08 09-52-54

This is also the case for #637

**Comment by albert-github** [comment]
This is strange.

I built the documentation with the current doxygen master (1.9.5 (35711f2d14acc81cff83f81f04c566ad8c8c6fe8)) for the current Gudhi master, the proposed pull request and the current Gudhi master with the changes for the proposed pull request and in all cases I got the drop down menu (Tested it locally on my Windows machine).

**Comment by VincentRouvreau** [comment]
> This is strange.
> 
> I built the documentation with the current doxygen master (1.9.5 (35711f2d14acc81cff83f81f04c566ad8c8c6fe8)) for the current Gudhi master, the proposed pull request and the current Gudhi master with the changes for the proposed pull request and in all cases I got the drop down menu (Tested it locally on my Windows machine).

CI is using doxygen 1.8.17 and I have the same behaviour locally with 1.9.2 version

**Comment by albert-github** [comment]
I've done "some" investigations.

The problem is still present when using the 1.9.2 version but is gone when using the 1.9.3 version.

I concentrated a bit on the 1.9.2 version and the only differences that are present here are in the HTML files:
```
35c35,36
< <link href="stylesheet.css" rel="stylesheet" type="text/css" />
---
> <link href="doxygen.css" rel="stylesheet" type="text/css" />
> <link href="stylesheet.css" rel="stylesheet" type="text/css"/>
234c235
<         Generated on Wed Jun 8 2022 11:16:32 for GUDHIdev by
---
>         Generated on Wed Jun 8 2022 11:04:20 for GUDHIdev by
```
these changes should all be harmless.

Furthermore there is the difference between the `stylesheet.css` used by default by Gudhi and the version of the `doxygen.css` plus the `stylesheet.css` as used by the proposed patch.

The Gudhi `stylesheet.css` is based on the doxygen 1.8.6 version and the version from the proposed patch are, of course, based on the used doxygen version (so they can use the new features of doxygen and the improvements in the layout.
When placing back the `doxygen.css` of the 1.8.6 version the patched version works again, so do the versions til the 1.8.14 doxygen version. The `doxygen.css` from the version 1.8.15 is the first `doxygen.css` that gives the shown problems (when running with doxygen <= 1.9.2).

When we look at the differences between the 1.8.14 and 1.8.15 `doxygen.css` we see: [diff_css_14_15.txt](https://github.com/GUDHI/gudhi-devel/files/8860432/diff_css_14_15.txt)

I looked at the differences of the `doxygen.css` of the 1.8.14 version and the 1.8.15 version and after some experiments the lines:
```
ul {
  overflow: hidden; /*Fixed: list item bullets overlap floating elements*/
}
```
cause the problems and these have been reset in the 1.9.3 version by:
```
ul {
  overflow: visible;
}
```
due to the pull request:
```
Commit: 590198b416cd53313d150428d2f912586065ea0d [590198b]
Date: Wednesday, December 1, 2021 1:37:26 PM
issue #8924 Horizontal scroll bar missing in HTML for wide class="dotgraph" objects
```

Conclusions:
- it is better that everywhere the newer doxygen version (>= 1.9.3) is used so one can benefit from the improvements (best solution)
- I'll  update this proposed pull request and the proposed pull request #637 with the corrected `ul` line (just a workaround to work with the older versions)

**Comment by VincentRouvreau** [comment]
> I've done "some" investigations.

Thank you very much for the investigation !
I will have to update doxygen on the docker container for the CI to generate doxygen.
I will test your proposed solution, but that sounds good !!

**Comment by VincentRouvreau** [comment]
As doxygen 1.9.3 is now available with conda, I could test it and your fix is Ok. Maybe we should specify doxygen >= 0.9.3 in src/common/doc/installation.h.

**Comment by VincentRouvreau** [comment]
> I think you didn't mean `0.9.3` in your previous comment but `1.9.3`.

Yes, sorry, 1.9.3

> The requirement to use a recent version of doxygen is a good requirement.

:+1: 

> Note that the current doxygen version is 1.9.4, it is a pity that conda didn't also place the 1.9.4 version in their tools (1.9.4. was release on May 5, 2022).

This is a work in progress: https://github.com/conda-forge/doxygen-feedstock/pull/31

**Comment by albert-github** [comment]
I read with proposed pull request #637:
> This proposed pull request obsoletes https://github.com/GUDHI/gudhi-devel/pull/635.

so I think it is better to try #637 and forget about #635.

**Comment by VincentRouvreau** [comment]
#637 fixes the issue, so I close this one

**Pull request closed without merging**